src/domain/Session.js

```diff
@@ -126,9 +129,21 @@ export default class Session {
   }
 
   primaryContext(): ?string {
+    if (this.engineVersion) {
+      try {
```

Comment:
Is the try/catch required for a `parseFloat` ?
Could you extract it to a method an add unit test on it ?